generate-osm/db-utils/find-coordinates.go

```diff
@@ -0,0 +1,312 @@
+package dbUtils
+
+import (
+	"math"
+	"sort"
+	"strconv"
+	OSMUtil "transform-osm/osm-utils"
+
+	"github.com/pkg/errors"
+)
+
+type nodePair struct {
+	node1    *OSMUtil.Node
+	node2    *OSMUtil.Node
+	dist     float64
+	remDist1 float64
+	remDist2 float64
+}
+
+const EARTH_RADIUS_CONST = 6371.0
+
+var endReached = errors.New("reached end of track.")
+var junctionReached = errors.New("reached a junction, cannot proceed.")
+
+func findNewNode(
+	osmData *OSMUtil.Osm,
+	node1 *OSMUtil.Node,
+	node2 *OSMUtil.Node,
+	dist1 float64,
+	dist2 float64,
+) (*OSMUtil.Node, error) {
+
+	if dist1 == 0.0 {
+		return node1, nil
+	}
+
+	var node *OSMUtil.Node
+
+	up1, upDist1, down1, downDist1, err1 := findNodes(osmData, node1, dist1)
+	up2, upDist2, down2, downDist2, err2 := findNodes(osmData, node2, dist2)
+
+	if err1 != nil {
+		return nil, errors.Wrap(err1, "insufficient anchor: "+node1.Id)
+	}
+	if err2 != nil {
+		return nil, errors.Wrap(err2, "insufficient anchor: "+node2.Id)
+	}
+
+	if up1 == up2 || up1 == down2 {
+		node, err := OSMUtil.GetNodeById(osmData, up1)
+		return node, err
+	} else if down1 == up2 || down1 == down2 {
+		node, err := OSMUtil.GetNodeById(osmData, down1)
+		return node, err
+	} else {
+		node = getClosestMatch(osmData, up1, up2, down1, down2, upDist1, upDist2, downDist1, downDist2)
+	}
+	return node, nil
+
+}
+
+func getClosestMatch(
+	osmData *OSMUtil.Osm,
+	up1, up2, down1, down2 string,
+	upDist1, upDist2, downDist1, downDist2 float64,
+) *OSMUtil.Node {
+	upNode1, _ := OSMUtil.GetNodeById(osmData, up1)
+	upNode2, _ := OSMUtil.GetNodeById(osmData, up2)
+	downNode1, _ := OSMUtil.GetNodeById(osmData, down1)
+	downNode2, _ := OSMUtil.GetNodeById(osmData, down2)
+
+	upNode1Lat, _ := strconv.ParseFloat(upNode1.Lat, 64)
+	upNode1Lon, _ := strconv.ParseFloat(upNode1.Lon, 64)
+	downNode1Lat, _ := strconv.ParseFloat(downNode1.Lat, 64)
+	downNode1Lon, _ := strconv.ParseFloat(downNode1.Lon, 64)
+
+	upNode2Lat, _ := strconv.ParseFloat(upNode2.Lat, 64)
+	upNode2Lon, _ := strconv.ParseFloat(upNode2.Lon, 64)
+	downNode2Lat, _ := strconv.ParseFloat(downNode2.Lat, 64)
+	downNode2Lon, _ := strconv.ParseFloat(downNode2.Lon, 64)

```

Comment:
These calls are basiclly the same. Can you add a method 
`func computeNodeInformation(osmData *OSMUtil.Osm, kilo float64) (node *OSMUtil.Node, nodeLat float64, nodeLon float64)`